beater/config/config.go

```diff
@@ -39,6 +42,11 @@ const (
 	msgInvalidConfigAgentCfg = "invalid value for `apm-server.agent.config.cache.expiration`, only accepting full seconds"
 )
 
+var (
```

Comment:
Actually now that I think of it, it's allowed to have neither - sorry. That would match all services, and would have lowest precedence.

Reply:
In which case we return the first agent_config?

Reply:
Here's how it should work. Return the first matching AgentConfig in the list, evaluating rules in order of precedence:

1. AgentConfig with both service name and environment set, matching the queried service name and environment
2. AgentConfig with only service name set, matching the queried service name
3. AgentConfig with only service environment set, matching the queried service environment
4. AgentConfig with neither service name nor environment set

If none of those resolve, return an empty result.

That follows what we do in the UI. You can select "Any" for service name and/or environment in the UI, which is equivalent to leaving that property unset in the agent_config entry.

Reply:
Alright, think I (hopefully) got it now